test/unit-tests/ooyala_ssai_test.js

```diff
@@ -292,17 +293,39 @@ describe('ad_manager_ooyala_ssai', function()
       ad: {}
     };
 
-    OoyalaSsai.onResponse(OoyalaSsai.currentId3Object, ssaiXml);
+    OoyalaSsai.onResponse(currentId3Object, ssaiXml);
 
     var ad = adQueue[0];
-
     // impression, and start tracking events
     OoyalaSsai.playAd(ad);
     expect(trackingUrlsPinged.startUrl).to.be(1);
     expect(trackingUrlsPinged.startUrl2).to.be(1);
     expect(trackingUrlsPinged.impressionUrl).to.be(1);
     expect(trackingUrlsPinged.impressionUrl2).to.be(1);
 
+    // pause and resume events
+    OoyalaSsai.pauseAd(ad);
+    expect(trackingUrlsPinged.pauseUrl).to.be(1);
+    OoyalaSsai.resumeAd(ad);
+    expect(trackingUrlsPinged.resumeUrl).to.be(1);
+
+    // mute and unmute events
+    OoyalaSsai.onMuteStateChanged("muteStateChanged", true);
+    expect(trackingUrlsPinged.muteUrl).to.be(1);
+    OoyalaSsai.onMuteStateChanged("muteStateChanged", false);
+    expect(trackingUrlsPinged.unmuteUrl).to.be(1);
+
+    // volume change events should also trigger mute / unmute urls
+    OoyalaSsai.onAdVolumeChanged("adVolumeChanged", 0);
+    expect(trackingUrlsPinged.muteUrl).to.be(2);
+    OoyalaSsai.onAdVolumeChanged("adVolumeChanged", 50);
+    expect(trackingUrlsPinged.unmuteUrl).to.be(2);
```

Comment:
Will a volume change that unmutes the player cause the unmuteUrl to be pinged twice? Can this happen with our player? We probably don't want the mute/unmuteUrls to be pinged twice on one mute state change.

Reply:
@aeng7 there is logic in the plugin that keeps track of the mute state and only pings the url if a change of state is found. This is in both `onMuteStateChanged` and `onAdVolumeChanged`, although there looks like a possible but in `onAdVolumeChanged` that does not check the `isMuted` bool on setting the volume to 0 so I will change that

Reply:
@aeng7 made some changes to the logic to ensure `lastVolume` and `isMuted` are consistent and added some unit tests to check and make sure the urls aren't pinged multiple times incorrectly